presto-common/src/main/java/com/facebook/presto/common/type/BigintEnumType.java

```diff
@@ -157,7 +162,7 @@ public boolean equals(Object o)
             return false;
         }
 
-        LongEnumType other = (LongEnumType) o;
+        BigintEnumType other = (BigintEnumType) o;
 
         return Objects.equals(getTypeSignature().getBase(), other.getTypeSignature().getBase())
                 && Objects.equals(getEnumMap(), other.getEnumMap());
```

Comment:
Is case sensitivity taken care of here?

Reply:
Hmm, the enum mapping should be case sensitive. The type name probably should not.